src/mococrw/dilithium.h

```diff
@@ -0,0 +1,392 @@
+     * supported types.
+     * @return the type of the asymmetric key.
+     * @throws This method may throw an OpenSSLException if OpenSSL indicates an error
+     */
+    AsymmetricKey::KeyTypes getType() const { return _key.getType(); }
+
+    /**
+     * Gets the specification of the asymmetric key in usage
+     *
+     * @return the specification of the key in use.
+     * @throws This method may throw an OpenSSLException if OpenSSL indicates an error
+     */
+    std::unique_ptr<DilithiumAsymmetricKey::Spec> getKeySpec() const { return _key.getKeySpec(); }
+
+    /**
+     * Gets the size of the Asymmetric key in bits
+     * @return the size of the key in bits
+     */
+    int getKeySize() const { return _key.getKeySize(); }
+
+    inline bool operator==(const DilithiumAsymmetricPublicKey &rhs) const
+    {
+        return *_internal() == *rhs._internal();
+    }
+    inline bool operator!=(const DilithiumAsymmetricPublicKey &rhs) const
+    {
+        return !(*this == rhs);
+    }
+
+protected:
+    DilithiumAsymmetricPublicKey(DilithiumAsymmetricKey &&key) : _key{std::move(key)} {}
+
+    DilithiumAsymmetricKey _key;
+};
+
+/**
+ * @brief Dilithium representation of AsymmetricKeypair
+ *
+ * @see AsymmetricKeypair
+ *
+ * This shall be replaced by AsymmetricKeypair once Dilithium is supported by the openssl version
+ * used by MoCOCrW.
+ * The interface is similar to the one of AsymmetricKeypair.
+ *
+ * For documentation see AsymmetricKeypair
+ */
+class DilithiumAsymmetricKeypair : public DilithiumAsymmetricPublicKey
+{
+public:
+    DilithiumAsymmetricKeypair(std::shared_ptr<DilithiumKeyImpl> keypair)
+            : DilithiumAsymmetricPublicKey{std::move(keypair)}
+    {
+    }
+
+    /**
+     * @brief Parses an ASN.1 struct in DER format at returns a private key object
+     *
+     * @param asn1Data The ASN.1 struct in DER format
+     * @return the DilithiumAsymmetricKeypair object created form the DER data
+     * @throws MoCOCrWException if the struct cannot be parsed or if the key is invalid
+     */
+    static DilithiumAsymmetricKeypair readPrivateKeyfromDER(const std::vector<uint8_t> &asn1Data);
+
+    /**
+     * Generate an asymmetric keypair with given Spec.
+     *
+     * @see DilithiumSpec
+     *
+     * @throws This method may throw an OpenSSLException if OpenSSL
+     *      indicates an error
+     */
+    static DilithiumAsymmetricKeypair generate(const DilithiumAsymmetricKey::Spec &spec);
+
+private:
+    DilithiumAsymmetricKeypair(DilithiumAsymmetricKey &&key)
+            : DilithiumAsymmetricPublicKey{std::move(key)}
+    {
+    }
+};
+
+/**
+ * For readability, this alias may sometime prove useful.
+ */
+using DilithiumAsymmetricPrivateKey = DilithiumAsymmetricKeypair;
+
+/**
+ * @brief Specification class for Dilithium keys (clone of AsymmetricKey::Spec)
+ *
+ * @see AsymmetricKey::Spec for mor information
+ *
+ */
+class DilithiumAsymmetricKey::Spec
```

Comment:
Have the `DilithiumSpec` as a standalone class and make it derived from `AsymmetricKey::Spec` in the future? Here, it is just boilerplate and doesn't serve any real purpose?